### PR TITLE
Support scaling for shardingsphere

### PR DESCRIPTION
**Pull request by avalon566**
For #3710.

### CODE REVIEW

File: sharding-scaling/sharding-scaling-core/src/main/java/org/apache/shardingsphere/shardingscaling/core/synctask/realtime/RealtimeDataSyncTask.java [hunk 1]
```diff
@@ -78,6 +78,7 @@ public void prepare() {
 
     @Override
     public void start(final ReportCallback callback) {
+        syncConfiguration.getReaderConfiguration().setTableNameMap(syncConfiguration.getTableNameMap());
```
Comment (KomachiSion) [inline sharding-scaling/sharding-scaling-core/src/main/java/org/apache/shardingsphere/shardingscaling/core/synctask/realtime/RealtimeDataSyncTask.java]: I think we can move the logic out to Controller.

File: sharding-scaling/sharding-scaling-core/src/main/java/org/apache/shardingsphere/shardingscaling/core/synctask/history/HistoryDataSyncTask.java [hunk 2]
```diff
@@ -108,6 +108,7 @@ private void getEstimatedRows() {
     }
     
     private void instanceSyncRunner() {
+        syncConfiguration.getReaderConfiguration().setTableNameMap(syncConfiguration.getTableNameMap());
```
Comment (KomachiSion) [inline sharding-scaling/sharding-scaling-core/src/main/java/org/apache/shardingsphere/shardingscaling/core/synctask/history/HistoryDataSyncTask.java]: I think we can move the logic out to Controller.

File: sharding-scaling/sharding-scaling-core/src/main/java/org/apache/shardingsphere/shardingscaling/core/config/yaml/YamlProxyRuleConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+import java.util.Map;
+
+/**
+ * Rule configuration for YAML.
+ *
+ * @author avalon566
+ */
+@Getter
+@Setter
+public final class YamlProxyRuleConfiguration implements YamlConfiguration {
```
Comment (KomachiSion) [inline sharding-scaling/sharding-scaling-core/src/main/java/org/apache/shardingsphere/shardingscaling/core/config/yaml/YamlProxyRuleConfiguration.java]: I think scaling do not need proxy full configuration. 
Using datasource configuration and rule configuration in zk will be better.

File: sharding-scaling/sharding-scaling-core/src/main/java/org/apache/shardingsphere/shardingscaling/core/synctask/realtime/RealtimeDataSyncTask.java (hunk 1 above)
Comment (avalon566) [reply sharding-scaling/sharding-scaling-core/src/main/java/org/apache/shardingsphere/shardingscaling/core/synctask/realtime/RealtimeDataSyncTask.java]: I will refactor configuration code.

File: sharding-scaling/sharding-scaling-core/src/main/java/org/apache/shardingsphere/shardingscaling/core/synctask/history/HistoryDataSyncTask.java (hunk 2 above)
Comment (avalon566) [reply sharding-scaling/sharding-scaling-core/src/main/java/org/apache/shardingsphere/shardingscaling/core/synctask/history/HistoryDataSyncTask.java]: I will refactor configuration code.

**Comment by KomachiSion** [comment]
@avalon566 CI failed, please solve it first.

**Pull request merged**